File: lib/src/modules/profile/dtos/update_user_request.dart
import 'dart:io';

import 'package:sneakerx/src/models/enums/user_status.dart';

class UpdateUserRequest {
  final int userId;
  final String username;
  final String fullName;
  final String email;
  final String phone;
  final String status;
  final String? avatarUrl;

  UpdateUserRequest({
    required this.userId,
    required this.username,
    required this.fullName,
    required this.email,
    required this.phone,
    required this.status,
    this.avatarUrl,
  });

  Map<String, dynamic> toJson() {
    return {
      'userId': userId,
      'username': username,
      'fullName': fullName,
      'email': email,
      'phone': phone,
      'status': status,
      'avatarUrl': avatarUrl,
    };
  }

}

File: src/main/java/com/example/sneakerx/dtos/user/UpdateUserRequest.java
package com.example.sneakerx.dtos.user;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.web.multipart.MultipartFile;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class UpdateUserRequest {
    private Integer userId;
    private String username;
    private String fullName;
    private String email;
    private String phone;
    private String status;
    private MultipartFile avatar;
}
